### PR TITLE
Run result exports from archives

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/archives/models.py [hunk 1]
```diff
@@ -86,5 +91,20 @@ def get_download_link(self):
         else:
             return ""
 
+    def iter_records(self):
+        """
+        Creates an iterator for the records in this archive, streaming and decompressing on the fly
+        """
+        s3 = self.s3_client()
+        s3_obj = s3.get_object(**self.s3_location())
+        stream = gzip.GzipFile(fileobj=s3_obj["Body"])
```
Comment (nicpottier) [inline temba/archives/models.py]: Is "Body" a stream or the full body?

Comment (rowanseymour) [reply temba/archives/models.py]: a stream

File: temba/flows/models.py [hunk 2]
```diff
@@ -5382,127 +5397,188 @@ def write_export(self):
...
+        print(f"Results export #{self.id} for org #{self.org.id}: found {len(run_ids)} runs in database to export")
 
         for id_batch in chunk_list(run_ids, 1000):
             run_batch = (
                 FlowRun.objects.filter(id__in=id_batch, contact__is_test=False)
-                .select_related("contact")
-                .prefetch_related("contact__all_groups")
-                .order_by("id")
+                .select_related("contact", "flow")
+                .order_by("modified_on")
```
Comment (rowanseymour) [inline temba/flows/models.py]: note to self - index?

File: temba/flows/models.py [hunk 3]
```diff
@@ -5382,127 +5397,200 @@ def write_export(self):
...
+        for archive in archives:
+            for record_batch in chunk_list(archive.iter_records(), 1000):
+                matching = []
+                for record in record_batch:
+                    if record["flow"]["uuid"] in flow_uuids and (not responded_only or record["responded"]):
+                        matching.append(record)
+                yield matching
+
+        # secondly get runs from database
+        runs = FlowRun.objects.filter(flow__in=flows).order_by("modified_on")
```
Comment (rowanseymour) [inline temba/flows/models.py]: we have some indexes on flow for the API, but unfortunately they use the opposite ordering:

```
"flows_flowrun_flow_modified_id" btree (flow_id, modified_on DESC, id DESC)
"flows_flowrun_flow_modified_id_where_responded" btree (flow_id, modified_on DESC, id DESC) WHERE responded = true
```

These get used but the run ids have to be sorted in memory, so need to decide if its worth adding new indexes to flows_flowrun.

**Comment by ericnewcomer** [comment]
⭐️

**Pull request merged**